fedimint-server/src/config/api.rs

```diff
@@ -388,6 +399,86 @@ impl ConfigGenApi {
     fn bad_request<T>(msg: &str) -> ApiResult<T> {
         Err(ApiError::bad_request(msg.to_string()))
     }
+
+    pub async fn restart_federation_setup(&self) -> ApiResult<()> {
+        let leader = {
+            let expected_status = [
+                ServerStatus::SharingConfigGenParams,
+                ServerStatus::ReadyForConfigGen,
+                ServerStatus::ConfigGenFailed,
+                ServerStatus::VerifyingConfigs,
+                ServerStatus::VerifiedConfigs,
+            ];
+            let mut state = self.require_status_maybe(&expected_status).map_err(|_| {
+                ApiError::bad_request(format!(
+                    "Expected to be in one of {:?} states",
+                    expected_status
+                ))
+            })?;
+
+            self.remove_configs()?;
+            state.status = ServerStatus::SetupRestarted;
+            info!(
+                target: fedimint_logging::LOG_NET_PEER_DKG,
+                "Update config gen status to 'Setup restarted'"
+            );
+            // Create a WSClient for the leader
+            state
+                .local
+                .clone()
+                .and_then(|local| local.leader_api_url.map(WsAdminClient::new))
+        };
+
+        self.update_leader().await?;
+
+        // Followers wait for leader to signal that all peers have restarted setup
+        // The leader will signal this by setting it's status to AwaitingPassword
+        let self_clone = self.clone();
+        let mut sub_group = self.task_group.make_subgroup().await;
+        sub_group
+            .spawn("restart", move |_handle| async move {
+                if let Some(client) = leader {
+                    // Followers wait for leader to signal that all peers have restarted setup
+                    loop {
+                        let status = client.status().await.map_err(|_| {
+                            ApiError::not_found("Unable to connect to the leader".to_string())
+                        })?;
```

Comment:
Shouldn't we keep re-trying if there is a network error?

Reply:
allowed for 3 retries